packages/flutter_tools/lib/src/ios/devices.dart

```diff
@@ -260,7 +260,7 @@ class IOSDevice extends Device {
           forwardDiagPort = _acquireAndForwardPort(ProtocolDiscovery.kDiagnosticService,
                                                    debuggingOptions.diagnosticPort);
         } else {
-          forwardDiagPort = new Future.value(null);
+          forwardDiagPort = new Future<int>.value(null);
```

Comment:
It does now. An older version of the analyzer had a bug and failed to catch this. We started seeing failures after we picked up a new analyzer.